backend/sql/initial_schema.sql

```diff
@@ -0,0 +1,100 @@
+create table project
+(
+    project_id  bigserial     not null primary key,
+    name        varchar(256)  not null unique,
+    description varchar(1000) null
+);
+
+create table test
+(
+    test_id bigserial    not null primary key,
+    name    varchar(256) not null unique
+);
+
+create table subtest
+(
+    subtest_id      bigserial      not null primary key,
+    test_id         bigint         not null,
+    name            varchar(256)   not null unique,
+    -- Method
+    meth            varchar(10000) null,
+    confluence_link varchar(1000)  null,
+    constraint belongs_to_test
+        foreign key (test_id) references test (test_id)
+            on update cascade
+            on delete cascade
+);
+
+-- Corresponds to an EXCEL file
+create table experiment
+(
+    experiment_id bigserial                not null primary key,
+    project_id    bigint                   not null,
+    subtest_id    bigint                   not null,
+    -- This is input from the user
+    name          varchar(256)             null,
+    meta          varchar(1000)            null,
+    -- This is fallback name to show if name is not specified
+    raw_file_name varchar(256)             not null,
+    upload_date   timestamp with time zone not null,
+
+    constraint belongs_to_project
+        foreign key (project_id) references project (project_id)
+            on update cascade
+            on delete cascade,
+
+    constraint has_subtest
+        foreign key (subtest_id) references subtest (subtest_id)
+            on update cascade
+            -- Prevent removing subtest if there is an experiment having such subtest
+            on delete no action
+);
+
+create table measurement
+(
+    measurement_id bigserial        not null primary key,
+    experiment_id  bigint           not null,
+    -- Either compound id or name
+    compound_id    varchar(256)     not null,
+    concentration  double precision not null,
```

Comment:
where is repetition in this schema? I mean we have this three sheets which form triple (compound, concentration signal), repetition will be in case of same compound and concentration, it's not explicitly given to us. well, we can check that there are same pairs of compound and concentration but signal might be a little bit different for them, what repetition will be equal to in this case? ok, we can have completely equal triples and then repetition will be in this case?

I am not sure I understand what controls are maybe it's the source of my confusion.

Reply:
If you look at the initial data file that they sent to us on notion, you will see that every compound have three repetitions (last segment of data with `1/3 2/3 3/3` and the last (row `K`) is control which just shows for some known drug (and some known concentration of active component) yields some *nominal* signal, which serves as an evidence that the experiment is OK.

In this schema, repetitions are horisontal groups of three lines. Your readouts can be something like

```
923 1456 2021 4993 17989 ...
821 1641 1999 5010 16595 ...
904 1399 1989 4892 17031 ...
...
```

The farther to the right, the less is concentration in this case. I'm not too certain how important it is to call it exactly `repetitions`, but it's a standard name in data science. 

I guess, for MVP you can ignore it completely, but it will cause bad UX and someone will ask to add it down the line. There is no reason to not store repetitions per compound per concentration for easier life of users and operators.

Reply:
@pva701 if there are triples `(lsd, 100, 500)` and `(lsd, 100, 501)`, we can say that the first one has `repetition = 1` and the second one has `repetition = 2`. If it's a standard name in data science and @serokellcao thinks it can be helpful to users, let's add it, it should be easy, right?

Reply:
We decided that it's unnecessary.

Reply:
Since we are not going to display this data (repetition number), it's probably useless to store it.